src/main/java/com/ncc/aif/ValidateAIFCli.java

```diff
@@ -0,0 +1,649 @@
+                    " (" + ++fileNum + " of " + filesToValidate.size() + ").");
+            final Model dataToBeValidated = ModelFactory.createDefaultModel();
+            boolean notSkipped = ((restriction != ValidateAIF.Restriction.NIST_TA3) || checkHypothesisSize(fileToValidate))
+                    && loadFile(dataToBeValidated, fileToValidate);
+            if (notSkipped) {
+                if (profiling) {
+                    stats.startCollection();
+                }
+                if (useProgressMonitor) {
+                    String filename = fileToValidate.getName().replace(".ttl", "") + "-progress.tab";
+                    ProgressMonitor pm;
+                    try {
+                        pm = new AIFProgressMonitor(filename);
+                    } catch (IOException e) {
+                        pm = new AIFProgressMonitor();
+                        logger.warn("Could not open progress monitor filename {}.  Writing progress to StdOut.", filename);
+                    }
+                    validator.setProgressMonitor(pm);
+                }
+                final Resource report = validator.validateKBAndReturnReport(dataToBeValidated);
+                if (profiling) {
+                    stats.endCollection();
+                    stats.dump(fileToValidate.toString());
+                }
+                if (report == null) {
+                    logger.warn("---> Could not validate " + fileToValidate + " (engine error).  Skipping.");
+                    skipCount++;
+                } else if (!ValidateAIF.isValidReport(report)) {
+                    invalidCount++;
+                    final int numViolations = processReport(report, fileToValidate, outputToFile);
+                    if (numViolations == maxValidationErrors || report.hasProperty(ThreadedValidationEngine.SH_ABORTED)) {
+                        logger.warn("---> Validation of " + fileToValidate +
+                                " was aborted after " + numViolations + " SHACL violations.");
+                        abortCount++;
+                    } else {
+                        logger.warn("---> Validation of " + fileToValidate + " failed.");
+                    }
+                }
+                date = Calendar.getInstance().getTime();
+                logger.info("---> completed " + format.format(date) + ".");
+            } else
+                skipCount++;
+
+            dataToBeValidated.close();
+        }
+
+        final ReturnCode returnCode = displaySummary(fileNum + nonTTLcount, invalidCount, skipCount + nonTTLcount, abortCount);
+        return returnCode.ordinal();
+    }
+
+    //TODO: make ArgGroup when 4.0 is stable
+    private void checkOntMutex() {
+        // Enforce mutual exclusion for domain ontologies
+        boolean hasCustom = customOntologies != null;
+        if (!(useProgramOntology || useLDCOntology || hasCustom)) {
```

Comment:
Does picocli not support this out of the box through configurations?

Reply:
4.0 supports it, but it's currently in Alpha and has a bug where the exclusivity is caught, but the wrong error message is produced. Hopefully, we'll change all these to ArgGroups when 4.0 is stable